x-pack/plugins/lens/public/datatable_visualization/expression.tsx

```diff
@@ -144,23 +189,86 @@ function DatatableComponent(props: DatatableProps & { formatFactory: FormatFacto
         columns={props.args.columns.columnIds
           .map(field => {
             const col = firstTable.columns.find(c => c.id === field);
+            const colIndex = firstTable.columns.findIndex(c => c.id === field);
+
+            const filterable = col?.meta?.type && props.getType(col.meta.type)?.type === 'buckets';
             return {
               field,
               name: (col && col.name) || '',
+              render: (value: unknown) => {
+                const formattedValue = formatters[field]?.convert(value);
+                if (filterable && value) {
+                  return (
+                    <EuiFlexGroup
+                      className="lnsDataTable__cell"
+                      data-test-subj="lnsDataTableCellValueFilterable"
+                      gutterSize="xs"
+                    >
+                      <EuiFlexItem grow={false}>{formattedValue}</EuiFlexItem>
```

Comment:
You've chosen not to put `grow={true}` on the main contents, which looks different from the Visualize Data Table. Why is this?

Reply:
Speaking with @cchaos , we've decided for aligning with the inspector look.

Reply:
It keeps the filter actions closer to the actual value (the text) which helps in not requiring a long distance to travel between value and actions.